data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducer.java

```diff
@@ -69,28 +72,39 @@ public class KafkaCustomProducer<T> {
 
     private final SchemaService schemaService;
 
+    private final KafkaTopicProducerMetrics topicMetrics;
 
-    public KafkaCustomProducer(final Producer producer,
+
+    public KafkaCustomProducer(final KafkaProducer producer,
                                final KafkaProducerConfig kafkaProducerConfig,
                                final DLQSink dlqSink,
                                final ExpressionEvaluator expressionEvaluator,
-                               final String tagTargetKey
+                               final String tagTargetKey,
+                               final KafkaTopicProducerMetrics topicMetrics
     ) {
         this.producer = producer;
         this.kafkaProducerConfig = kafkaProducerConfig;
         this.dlqSink = dlqSink;
-        bufferedEventHandles = new LinkedList<>();
+        this.bufferedEventHandles = new LinkedList<>();
         this.expressionEvaluator = expressionEvaluator;
         this.tagTargetKey = tagTargetKey;
         this.topicName = ObjectUtils.isEmpty(kafkaProducerConfig.getTopic()) ? null : kafkaProducerConfig.getTopic().getName();
         this.serdeFormat = ObjectUtils.isEmpty(kafkaProducerConfig.getSerdeFormat()) ? null : kafkaProducerConfig.getSerdeFormat();
-        schemaService = new SchemaService.SchemaServiceBuilder().getFetchSchemaService(topicName, kafkaProducerConfig.getSchemaConfig()).build();
+        this.schemaService = new SchemaService.SchemaServiceBuilder().getFetchSchemaService(topicName, kafkaProducerConfig.getSchemaConfig()).build();
+        this.topicMetrics = topicMetrics;
+        this.topicMetrics.register(this.producer);
+    }
+
+    KafkaTopicProducerMetrics getTopicMetrics() {
+        return topicMetrics;
     }
 
     public void produceRawData(final byte[] bytes, final String key) {
         try {
             send(topicName, key, bytes).get();
+            topicMetrics.update(producer);
         } catch (Exception e) {
+            // TODO: Metrics
```

Comment:
These are still TODO?

Reply:
These are producer metrics for exception handling. Needs to be done as separate PR.